### PR TITLE
Added initial prototype of holocube package and notebook

### PR DESCRIPTION
**Pull request by philippjfr**
Initial prototype for cube-explorer project, includes small set of Element types and plots allowing basic plotting with cartopy and cube exploration with iris. The accompanying notebook goes through how to begin using these Elements to display custom projections and then lays out an initial workflow for working with Iris Cubes, using the example described in Issue #3. Still very much a work in progress but gives an idea of what plotting with Cartopy+Iris+HoloViews might look like and provides an initial implementation to wrap and convert iris cubes into individually displayable Elements.

Outstanding items:
- [x] Decide on name for package (sticking with holocube for now)
- [x] Decide on package structure (follow HoloViews structure for now)
- [x] Document classes and modules
- [x] Create unit test directory and write some tests (starting with Cube)
- [x] Set up travis

~~Note that this PR still depends on the a [philippjfr/iris](https://github.com/philippjfr/iris) fork of Iris ([PR #1942](https://github.com/SciTools/iris/pull/1942))~~ ~~It now also depends on [PR #542](https://github.com/ioam/holoviews/pull/542) in holoviews~~.

The PR now works with Iris and HoloViews master.

### CODE REVIEW

**Comment by jlstevens** [comment]
Great! 

It is worth mentioning that `holocube` is just a suggestion and we are happy to adopt any other suggestions. The only reason we didn't stick with cube-explorer is that `from cube-explorer import ...` is awkward/doesn't work. 

We could remove the hyphen but `cubeexplorer` isn't a great name either due to the repeated `e`...

**Comment by jbednar** [comment]
There's also cubebrowser, which works better than cubeexplorer; not sure what the distinction should be.  Or cubeviews. :-)

File: holocube/element/cube.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    kwargs = {}
+    if coord.units.is_time_reference():
+        kwargs['value_format'] = coord.units.num2date
+    else:
+        kwargs['unit'] = str(coord.units)
+    return Dimension(coord.name(), **kwargs)
+
+
+class Cube(Element):
+
```
Comment (jbednar) [inline holocube/element/cube.py]: Docstrings, here and elsewhere?

File: holocube/element/cube.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                        raise ValueError('Key dimension %s not found in Iris cube.' % kd)
+                    coords.append(coord[0])
+            else:
+                coords = data.dim_coords
+            params['kdims'] = [coord_to_dimension(coord)
+                               for coord in coords]
+            if 'vdims' not in params:
+                params['vdims'] = [Dimension(data.name(), unit=str(data.units))]
+        else:
+            raise ValueError('Cube data must be of Iris Cube type.')
```
Comment (jbednar) [inline holocube/element/cube.py]: I personally prefer to reverse the order of checks like this, checking datatype at the start and raising the error right away.  That way the rest of the function won't need indenting, which avoids requiring the user to put it all in a mental stack buffer.

**Comment by marqh** [comment]
ok, i think we are happy with this as an initial cut

the top priority for me is to get this onto master and work with smaller changes, with that in mind:
go go gadget merge :)

**Comment by philippjfr** [comment]
Yay, thanks! Will start submitting small issues and PRs for improvements now.

**Comment by jlstevens** [comment]
Great! Now we should now be able to get Travis configured!

**Pull request merged**